src/command.rs

```diff
@@ -139,3 +167,89 @@ impl ShellCommand for ExternalCommand {
         Ok(child.stdout.take())
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct LlmCommand {
+    prompt: String,
+    openai_client: OpenAIClient,
+}
+
+impl LlmCommand {
+    pub fn new(prompt: String, openai_client: OpenAIClient) -> Self {
+        LlmCommand {
+            prompt,
+            openai_client,
+        }
+    }
+
+    pub async fn generate_response(&self, input: Option<String>) -> Result<String, Box<dyn Error>> {
+        let context = if let Some(input) = input {
+            format!("{}: {}", self.prompt, input)
+        } else {
+            self.prompt.clone()
+        };
+        let output = self.openai_client.generate_text(&context, 100).await?;
+        Ok(output)
+    }
+}
+
+impl Runnable for LlmCommand {
+    fn run(&self) -> Result<String, Box<dyn Error>> {
+        let runtime = Runtime::new().unwrap();
+        let output = runtime.block_on(self.generate_response(None))?;
+        Ok(output)
```

Comment:
Is `runtime` needed because the http request is async?

Reply:
Apparently so, tried removing it but fails as self.generate_response returns a `future` because it is an async function.